hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/lock/OzoneLockProvider.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.hadoop.ozone.om.lock;
+
+import org.apache.hadoop.ozone.om.helpers.BucketLayout;
+
+/**
+ * OzoneLockProvider class returns the appropriate lock strategy pattern
+ * implementation based on the configuration flags passed.
+ */
+public class OzoneLockProvider {
```

Comment:
Please add a test case to verify the different strategies and acquire/release locks.

OBS - two locks BUCKET_LOCK and KEY_PATH_LOCK
Regular - only BUCKET_LOCK

Reply:
Added a UT for OzoneLockProvider, thank you for the comment!